### PR TITLE
Actions

### PR DESCRIPTION
**Pull request by zhen0**
PR Checklist:

- [ ] add a short description of what's changed to the top of the `CHANGELOG.md`
- [ ] add/update tests (or don't, for reasons explained below)

### CODE REVIEW

File: src/pages/Dashboard/Actions/AddActionCard.vue [hunk 1]
```diff
@@ -34,11 +35,29 @@ export default {
         select: {
           title: '...'
         }
-      }
+      },
+      SLATypes: [
+        { name: 'starts but does not finish', enum: 'STARTED_NOT_FINISHED' },
+        {
+          name: 'is scheduled but does not start',
```
Comment (madkinsz) [inline src/pages/Dashboard/Actions/AddActionCard.vue:42]: Note that the check duration begins at the scheduled start time not the flow 'Scheduled' state transition time.

File: src/pages/Dashboard/Actions/AddActionCard.vue [hunk 2]
```diff
@@ -34,11 +35,29 @@ export default {
         select: {
           title: '...'
         }
-      }
+      },
+      SLATypes: [
+        { name: 'starts but does not finish', enum: 'STARTED_NOT_FINISHED' },
```
Comment (madkinsz) [inline src/pages/Dashboard/Actions/AddActionCard.vue:40]: Note that the check duration begins at the flow 'Running' state transition time.

Comment (madkinsz) [reply src/pages/Dashboard/Actions/AddActionCard.vue:40]: Might be helpful in some sort of tooltip?

File: src/pages/Dashboard/Actions/ActionCard.vue [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+      ></div
+    >
+    <div class="pl-8 pb-8 pt-0"
+      ><v-icon color="codePink" class="pr-2">{{
+        hookDetails[hook.event_type].icon
+      }}</v-icon
+      >When {{ hookType }}<span class="font-weight-bold">{{ hookName }}</span>
+      {{ hookDetail }}
+      <span v-if="includeSeconds">
+        for <span class="font-weight-bold">{{ seconds }} seconds</span> </span
```
Comment (madkinsz) [inline src/pages/Dashboard/Actions/ActionCard.vue:195]: "starts but does not finish after 60 seconds" makes more sense to me
perhaps
"starts but does not finish in 60 seconds" ?
```suggestion
        after <span class="font-weight-bold">{{ seconds }} seconds</span> </span
```

File: src/pages/Dashboard/Actions/ActionCard.vue [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+            ><v-icon>more_horiz</v-icon></v-btn
+          ></template
+        >
+        <v-card>
+          <div
+            ><v-btn
+              :style="{ 'text-transform': 'none', 'min-width': '0px' }"
+              text
+              width="100%"
+              color="secondaryGray"
```
Comment (whimsicallyson) [inline src/pages/Dashboard/Actions/ActionCard.vue:296]: check these in dark mode? utilGrayDark might be better

File: src/pages/Dashboard/Actions/AddActionCard.vue [hunk 5]
```diff
@@ -0,0 +1,930 @@
...
+</script>
+
+<template>
+  <v-card v-if="!addAction" outlined>
+    <v-card-title>
+      <v-spacer></v-spacer>
+      <v-btn
+        v-if="step != 'openAgentOrFlow' || hookDetail"
+        text
+        class="grey--text text--darken-2 light-weight-text mr-1"
```
Comment (whimsicallyson) [inline src/pages/Dashboard/Actions/AddActionCard.vue:544]: check these in dark mode too?

File: src/pages/Dashboard/Actions/AddActionCard.vue [hunk 6]
```diff
@@ -0,0 +1,930 @@
...
+      <!-- <div>
+        <v-chip
+          v-for="item in states['All']"
+          :key="item"
+          label
+          :disabled="disableClick"
+          max-width="300px"
+          class="ma-1"
+          outlined
+          :color="chosenStates.includes(item) ? 'pink' : ''"
```
Comment (whimsicallyson) [inline src/pages/Dashboard/Actions/AddActionCard.vue:814]: not codePink? (I swear I'm looking at more than colors lol)

File: src/pages/Dashboard/Actions/AddDoThis.vue [hunk 7]
```diff
@@ -0,0 +1,676 @@
...
+        ></span
+      >
+    </v-card-text>
+
+    <v-card-text v-if="step === 'selectMessageType'">
+      <v-row class="py-3">
+        <v-col v-for="type in actionTypes()" :key="type.title" cols="6" sm="2">
+          <div
+            v-ripple
+            class="chip-bigger d-flex align-center justify-start pa-2 cursor-pointer user-select-none"
```
Comment (whimsicallyson) [inline src/pages/Dashboard/Actions/AddDoThis.vue:395]: possibly round these corners, as we're discussing

File: src/pages/Dashboard/Actions/AddDoThis.vue [hunk 8]
```diff
@@ -0,0 +1,676 @@
...
+      <!-- <v-text-field
+        v-if="messageType.type === 'SLACK_WEBHOOK'"
+        v-model="messageConfigTo"
+        :label="messageConfigLabel"
+        :rules="[rules[messageType.type]]"
+        validate-on-blur
+        :error-messages="errorMessage"
+        @keyup.enter="saveConfig"
+      ></v-text-field> -->
+      <div v-else-if="isPagerDuty">
```
Comment (whimsicallyson) [inline src/pages/Dashboard/Actions/AddDoThis.vue:512]: Screen Shot 2021-03-22 at 5 35 52 PM
I think pagerduty's missing a step? (to whom?)

File: src/utils/cloudHooks.js [hunk 9]
```diff
@@ -91,3 +173,38 @@ export const STATES = {
...
+    'Resume',
+    'Queued',
+    'Submitted',
+    'Paused',
+    'Running',
+    'Finished',
+    'Success',
+    'Cancelled',
+    'Cached',
+    'Triggerfailed',
```
Comment (whimsicallyson) [inline src/utils/cloudHooks.js:192]: Can this be 2 words?

File: src/utils/cloudHooks.js [hunk 10]
```diff
@@ -91,3 +173,38 @@ export const STATES = {
...
+    'Skipped',
+    'Timedout',
```
Comment (whimsicallyson) [inline src/utils/cloudHooks.js:194]: this one as well

**Comment by zhen0** [comment]
Also just noting that we need to add a "You have no secrets yet" if there's no secrets for the secret name lists - especially slack webhooks.

**Comment by zhen0** [comment]
To keep things neater and reflect the new name, I'm replacing this draft PR with #695

**Pull request closed without merging**